plugin/storage/kafka/options.go

```diff
@@ -94,3 +181,36 @@ func (opt *Options) InitFromViper(v *viper.Viper) {
 func stripWhiteSpace(str string) string {
 	return strings.Replace(str, " ", "", -1)
 }
+
+// getCompressionLevel to get compression level from compression type
+func getCompressionLevel(compressionMode string, compressionLevel int) (int, error) {
+	compressionModeData := compressionModes[compressionMode]
+
+	if compressionLevel == defaultCompressionLevel {
```

Comment:
this is confusing, I suggest `if compressionLevel == 0 {`, which makes it obvious that default type's default value is being converted.

Reply:
This is a snippet from sarama library. this is the reason why I made defaultCompressionLevel = -1000 
```
// CompressionLevelDefault is the constant to use in CompressionLevel
// to have the default compression level for any codec. The value is picked
// that we don't use any existing compression levels.
const CompressionLevelDefault = -1000
```

Reply:
that's not the point. The meaning of "default" in sarama is different from what we have here. When dealing with CLI, if you want to use the default value for the respective compressor you simply not specify compression-level flag. Both 0 and -1000 are invalid values for most compressors except for zstd, where BOTH are valid and neither is the default. So I don't see a reason why this bad parameter design from sarama should be carried over to our flags. 0 is the natural null value for int field. It makes no sense to be playing tricks with -1000, which achieve the same results.

Reply:
I agree with you, i will make the changes

Reply:
It's been fixed.